src/runtime/schema.ts

```diff
@@ -57,10 +57,66 @@ export class Schema {
       }
     }
     if (options.description) {
+      // The descriptions should be passed ready for assignment into this.description.
+      // TODO(cypher1): Refactor the schema construction code to do this rearrangement at the call site.
       options.description.description.forEach(desc => this.description[desc.name] = desc.pattern || desc.patterns[0]);
     }
   }
 
+  private forEachRefinement<T, V>(func: (val: V) => T) {
```

Comment:
Does `func` need to have a return type? The result is being lost in the `forEach` loop.

Reply:
If not, the type could be a `hot`, `Consumer<V>`.

Reply:
So it could. I originally was passing the state through the func as well and then returning, will update to be a Consumer and use an explicit void to make it clearer.

Reply:
Also the body could be just two lines... but I'm not sure its readable. Thoughts?
```
    const types = [this, ...Object.values(this.fields)];
    types.forEach(type => type.refinement && func(type.refinement));
```

Reply:
LGTM